tests/infer/test_valid_models.py

```diff
@@ -859,6 +860,69 @@ def guide():
     assert_error(model, guide, TraceGraph_ELBO(), match='Shape mismatch inside plate')
 
 
+def test_unplate_name_ok():
+
+    def model():
+        a = pyro.sample("a", dist.Normal(0, 1))
+        assert a.shape == ()
+        with pyro.plate("plate", 2):
+            b = pyro.sample("b", dist.Normal(0, 1))
+            assert b.shape == (2,)
+            with unplate("plate"):
+                c = pyro.sample("c", dist.Normal(0, 1))
```

Comment:
The non-intuitive behavior I was concerned about is when the distribution here incorrectly depends on an enumerated variable inside `"plate"`. Do you think our existing enumeration checking logic in `TraceEnum_ELBO` is sufficient to help someone figure out that an `unplate` is breaking tractability?

Reply:
Yes, I believe any break in tractability possible with `block_plate` had already been addressed when we introduced reusable plates.